src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -3,46 +3,99 @@
 
 using System;
 using System.Net.Http;
-using System.Net.Http.Headers;
 using Microsoft.Identity.Client.AuthScheme.PoP;
 
 namespace Microsoft.Identity.Client.AppConfig
 {
 
     /// <summary>
-    /// Configuration properties used to construct a proof of possession request.
+    /// Details about the HTTP request and configuration properties used to construct a proof of possession request.
     /// </summary>
+    /// <remarks> 
+    /// POP tokens are signed by the process making the request. By default, MSAL will generate a key in memory.
+    /// To use a hardware key or an external key, implement <see cref="PopCryptoProvider"/>.
+    /// </remarks>
 #if !SUPPORTS_CONFIDENTIAL_CLIENT
     [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
 #endif
     public class PoPAuthenticationConfiguration
     {
         /// <summary>
-        /// Constructs the configuration properties used to construct a proof of possession request
+        /// Creates a configuration using the default key management. Use the properties <see cref="HttpMethod"/>, <see cref="HttpHost"/> etc. 
+        /// to control which elements of the request should be included in the POP token.
         /// </summary>
-        /// <param name="requestUri">An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri</param>
-        public PoPAuthenticationConfiguration(Uri requestUri)
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3 for details about signed HTTP requests.
+        /// </remarks>
+        public PoPAuthenticationConfiguration()
         {
             ConfidentialClientApplication.GuardMobileFrameworks();
-            RequestUri = requestUri ?? throw new ArgumentNullException(nameof(requestUri));
         }
 
         /// <summary>
-        /// An Uri to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request uri (to the host and query parts of it)
+        /// Creates a configuration using the default key management, and which binds all the details of the HttpRequestMessage.
         /// </summary>
-        public Uri RequestUri { get; }
+        /// <remarks>
+        /// Currently only the HtppMethod (m), UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(HttpRequestMessage httpRequestMessage)
```

Comment:
new ctor ... for full binding.

Reply:
We might also want to fix https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2285 at the same time?